tests/test_examples.py

```diff
@@ -809,10 +819,195 @@ def test_multiobject_optimization_prox():
+@pytest.mark.solve
+def test_omnigenity_qa():
+    """Test optimizing omnigenity parameters to match an axisymmetric equilibrium."""
+    # Solov'ev examples has B_max contours at theta=pi, need to change to theta=0
+    eq = get("SOLOVEV")
+    rone = np.ones_like(eq.R_lmn)
+    rone[eq.R_basis.modes[:, 1] != 0] *= -1
+    eq.R_lmn *= rone
+    zone = np.ones_like(eq.Z_lmn)
+    zone[eq.Z_basis.modes[:, 1] != 0] *= -1
+    eq.Z_lmn *= zone
+    lone = np.ones_like(eq.L_lmn)
+    lone[eq.L_basis.modes[:, 1] != 0] *= -1
+    eq.L_lmn *= lone
+    eq.axis = eq.get_axis()
+    eq.surface = eq.get_surface_at(rho=1)
+    eq.Psi *= 5  # B0 = 1 T
+    eq.solve()
+
+    field = OmnigenousField(
+        L_well=1,
+        M_well=4,
+        L_shift=1,
+        M_shift=1,
+        N_shift=1,
+        NFP=eq.NFP,
+        helicity=(1, 0),
+    )
+
+    eq_axis_grid = LinearGrid(rho=1e-2, M=4 * eq.M, N=4 * eq.N, NFP=eq.NFP, sym=False)
+    eq_lcfs_grid = LinearGrid(rho=1.0, M=4 * eq.M, N=4 * eq.N, NFP=eq.NFP, sym=False)
+
+    field_axis_grid = LinearGrid(
+        rho=1e-2, theta=2 * field.M_well, N=2 * field.N_shift, NFP=field.NFP, sym=False
+    )
+    field_lcfs_grid = LinearGrid(
+        rho=1.0, theta=2 * field.M_well, N=2 * field.N_shift, NFP=field.NFP, sym=False
+    )
+
+    objective = ObjectiveFunction(
+        (
+            Omnigenity(
+                field=field, eq=eq, eq_grid=eq_axis_grid, field_grid=field_axis_grid
+            ),
+            Omnigenity(
+                field=field, eq=eq, eq_grid=eq_lcfs_grid, field_grid=field_lcfs_grid
+            ),
+        )
+    )
+    constraints = (
+        FixModeR(eq=eq),
+        FixModeZ(eq=eq),
+        FixModeLambda(eq=eq),
+        FixPressure(eq=eq),
+        FixIota(eq=eq),
+        FixPsi(eq=eq),
+    )
+
+    optimizer = Optimizer("lsq-exact")
+    (eq, field), _ = optimizer.optimize(
+        (eq, field),
+        objective,
+        constraints,
+        maxiter=100,
+        ftol=1e-6,
+        xtol=1e-6,
+        verbose=3,
+    )
+
+    B_lm = field.B_lm.reshape((field.well_basis.L + 1, -1))
+    B0 = field.well_basis.evaluate(np.array([[0, 0, 0]])) @ B_lm
+    B1 = field.well_basis.evaluate(np.array([[1, 0, 0]])) @ B_lm
+
+    # x_lmn=0 because the equilibrium is QS
+    np.testing.assert_allclose(field.x_lmn, 0, atol=1e-12)
+
+    # check that magnetic well parameters get |B| on axis correct
+    grid = LinearGrid(N=eq.N_grid, NFP=eq.NFP, rho=0)
+    data = eq.compute("|B|", grid=grid)
+    np.testing.assert_allclose(B0, np.mean(data["|B|"]), rtol=1e-3)
+
+    # check that magnetic well parameters get |B| min & max on LCFS correct
+    grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP, rho=1)
+    data = eq.compute(["min_tz |B|", "max_tz |B|"], grid=grid)
+    np.testing.assert_allclose(np.min(B1), data["min_tz |B|"][0], rtol=2e-3)
+    np.testing.assert_allclose(np.max(B1), data["max_tz |B|"][0], rtol=2e-3)
+
+
+@pytest.mark.regression
+@pytest.mark.solve
```

Comment:
I added a new mark `optimize` recently that would be better here instead of `solve`